MuteToggler@jebeaudet.com/files/MuteToggler@jebeaudet.com/applet.js

```diff
@@ -192,12 +213,51 @@ MyApplet.prototype = {
         }
     },
 
+    evaluate_soundcard: function() {
+      // only use specific soundcard if searchstring is not empty
+      if (this.soundcard.trim() !== '')
+      {
+        // per default use first soundcard 
+        this.soundcard_id = "0";
+  
+        let soundcard_list_cmd = "cat /proc/asound/cards"; 
+        let [res, stdout] = GLib.spawn_command_line_sync(soundcard_list_cmd); 
+
+        if (res) {
+          // Split the result into lines 
+          let lines = stdout.toString().split('\n');
+
+          // Filter lines that contain the soundcard search string
+          let filteredLines = lines.filter(line => line.includes(this.soundcard)); 
+
+          // Extract the field after splitting by space and getting the second field (index 1) 
+          this.soundcard_id = filteredLines.map(line => line.split(' ')[1]);
+        }
+        log("Use specific soundcard id: '" + this.soundcard_id + "'");
+
+        this.amixer = this.amixer + " -c " + this.soundcard_id;
+      }
+    },
+
+    evaluate_input: function() {
+      this.input = POTENTIAL_INPUTS[0];
+      let [res, stdout] = GLib.spawn_command_line_sync(this.amixer);
```

Comment:
avoid using synchronous functions; prefer asynchronous functions.

Reply:
If using asynchronous functions: How can we be sure that evaluate_soundcard was already executed when executing evaluate_input?

Reply:
Welcome in the world of async, await, promise etc.

https://www.w3schools.com/js/js_async.asp

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

Reply:
now i did this via spawn_async. Any benefits of spawnCommandLineAsyncIO?

Reply:
Advantages, not really, but I think it's easier to use.